File: Data Structure and Algorithms/in_class/search_algorithms/linear_search.cpp
#include <iostream>
#include <vector>

using namespace std;

int linear_search(int array[], int index, int size) {

    for (int i = 0; i < size; i++) {
        if (array[i] == index) {
            return i;
        }
    }

    return -1;
}

int main() {
    int array[9];
    int size = sizeof(array) / sizeof(int);
    int index_to_find;

    for (int i = 0; i < size; i++) {
        cout << "Enter Number: ";
        cin >> array[i];
    }

    cout << endl;
    cout << "Enter Number to Find: ";
    cin >> index_to_find;
    int index_found = linear_search(array, index_to_find, size);

    if (index_found != -1) {
        cout << "Item Found: Index = " << index_found << endl;
    } else {
        cout << "Item not Found!" << endl;
    }

    return 0;
}

File: Data Structure and Algorithms/in_class/search_algorithms/linear_search.lua
function LinearSearch(Array, index)
    for i, v in pairs(Array) do
        if v == index then
            return i
        end
    end

    return -1
end


local Items = {
    gas = 5,
    book = 10,
}

local foundItem = LinearSearch(Items, 5)
print(foundItem ~= -1 and "found" or "not found")
print(foundItem)